helper/hex/hex_test.go

```diff
@@ -38,3 +39,26 @@ func TestDecodeUint64(t *testing.T) {
 		assert.Equal(t, uint64Array[index], decodedValue)
 	}
 }
+
+func TestDecodeHexToBig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Error on decoding", func(t *testing.T) {
+		t.Parallel()
+
+		_, err := DecodeHexToBig("012345q")
```

Comment:
Maybe you can assert that returned value is nil as well?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1289/commits/8513e9d2713d95a0bbb1df4eed52c8d0422cc6a0